### PR TITLE
If implemented, use ASCOM driver's LastExpsoure properties for image time and duration metadata

### PR DESCRIPTION
**Pull request by daleghent**
## 🚀 Purpose

Issue #309 raised the point that an ASCOM driver might reveal more accurate image exposure start time and duration information if the camera in use is equipped with a GPS, and if the driver does additional time-consuming tasks before the actual exposure commences. The Moravian ASCOM driver will populate `LastExposureStartTime` with GPS-sourced time value. Looking at the ZWO ASCOM driver, it will as well for some yet-to-be-released camera model that has a GPS.  Thinking more about this, some CCD drivers may do RBI flashing before the start of the exposure.

This PR alters the NINA camera ASCOM/Alpaca client to use `LastExposureStartTime` and `LastExposureDuration` if they are implemented. Their implemented presence is tested upon connection with the ASCOM/Alpaca driver and is used if they are. If not, the existing NINA-generated time and duration values are used.

## 🧪 How Was It Tested?

With the Alpaca camera OmniSim and the NINA simulator camera.

## 🤖 AI / Tooling Disclosure

None, as there is still some joy in creating small fixes like this.

## ✅ PR Checklist

- [x ] All unit tests pass
- [ ] UI changes tested across Light, Dark, and Night themes (if applicable)
- [x] Plugin API compatibility reviewed  
  - [ x] No breaking changes to interfaces  
  - [ x] No changes to method/class signatures (especially optional parameters)
- [ x] `RELEASE_NOTES.md` updated (if applicable)
- [ ] [Documentation](https://github.com/isbeorn/nina.docs) updated (if applicable)

## 🔗 Related Issues

#309

### CODE REVIEW

File: NINA.Equipment/Equipment/MyCamera/AscomCamera.cs [hunk 1]
```diff
@@ -106,6 +100,32 @@ private void Initialize() {
...
             }
 
+            // Determine LastExposure* Capabilities
+            // It seems that most drivers that do not implement these will throw NotImplementedException even when there is not yet an exposure.
+            // This is good, because it lets us know early if the properties are not available for use following an actual exposure being made.
+            // For drivers that do implement these properties but do not yet have an exposure to describe, they *should* throw InvalidOperationException
+            // but some don't and instead return bogus values such as an empty date string. We consider these to be available but they are obviously
+            // non-conformant with the ASCOM spec.
+            try {
+                _ = device.LastExposureDuration;
```
Comment (isbeorn) [inline NINA.Equipment/Equipment/MyCamera/AscomCamera.cs:110]: is this guaranteed to be present before having done any exposure?

File: NINA.Equipment/Equipment/MyCamera/AscomCamera.cs [hunk 2]
```diff
@@ -447,33 +467,32 @@ public IList<int> Gains {
...
-                        val = device.LastExposureStartTime;
+                DateTime? val = null;
+
+                if (ShouldBeConnected && HasLastExposureStartTime) {
+                    string startTimeStr = GetProperty<string>(nameof(Camera.LastExposureStartTime), null);
+
+                    if (DateTime.TryParse(startTimeStr, CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal, out DateTime dt)) {
+                        val = dt;
+                    } else {
+                        Logger.Error($"Unable to parse LastExposureStartTime '{startTimeStr}'");
```
Comment (isbeorn) [inline NINA.Equipment/Equipment/MyCamera/AscomCamera.cs:494]: could be spammy if it happens on every exposure

File: NINA.Equipment/Equipment/MyCamera/AscomCamera.cs (hunk 1 above)
Comment (daleghent) [reply NINA.Equipment/Equipment/MyCamera/AscomCamera.cs:110]: That's a good question, and one I spend a large amount of attention researching. There are no "HasLast.." booleans for these things so that knowledge of their implementation can be declared up-front; you have to use them and see if you catch a `PropertyNotImplemented` exception. In line with other ASCOM semantics, I would expect that a driver that truly does not implement these LastExposure properties to always throw a PNI on them, regardless of when they are accessed.

If the driver **does** implement them, getting the `LastExposure...` properties during our `Initialize()` in the camera client _should_ get an `InvalidOperationException` in response, which indicates they are implemented but have no values to provide yet as it's more likely that no exposures have been made under that instance of the driver yet.

There is the corner case of the Alpaca or local-server ASCOM driver already having done an exposure and thus has values to return during that test in `Initialize()`,  so that's accounted for as well. That would be we call their getter and get valid data back.

There is one additional case of non-conformance that I saw where a ZWO camera with GPS support can return an empty string instead of throw `InvalidOperationException` if no exposure has not yet been made. (reading through the driver code with ILSpy). Their date string is initialized as an empty string and there is no test for this case in their `LastExposureStartTime` getter, so it would just return the empty string. This is counted by the test as it being present, just in a non-conformant form prior to the first exposure being made by that instance.

Of course, like any ASCOM/Alpaca construct, a given driver's proper implementation of the expected semantics is anyone's guess. I would not be surprised if this change uncovers new ASCOM driver implementation bugs.

File: NINA.Equipment/Equipment/MyCamera/AscomCamera.cs (hunk 2 above)
Comment (daleghent) [reply NINA.Equipment/Equipment/MyCamera/AscomCamera.cs:494]: Could be. I can add a tracking variable to this so that the first _n_ instances of it occurring is logged but no more.

File: NINA.Equipment/Equipment/MyCamera/AscomCamera.cs (hunk 1 above)
Comment (isbeorn) [reply NINA.Equipment/Equipment/MyCamera/AscomCamera.cs:110]: Ok sounds good to me. Given that this can throw InvalidOperation or also return en empty string that would be reasonable.

File: NINA.Equipment/Equipment/MyCamera/AscomCamera.cs (hunk 2 above)
Comment (isbeorn) [reply NINA.Equipment/Equipment/MyCamera/AscomCamera.cs:494]: We can leave it like this for now and check which drivers do actually run into it. When we have a spammy log the chances are we will see it sooner.
